assimilation_code/modules/utilities/parse_args_mod.f90

```diff
@@ -24,52 +30,64 @@ module parse_args_mod
 !   a b c
 !   EOF
 !
-! OR:
+! to use from a file, open the file and:
 !
 !   read(unitnum, "(A256)") line
 !   call get_args_from_string(line, wordcount, words)
 !
 ! the limit on the number of words and the length of each string
 ! is determined by the 'words' character array that is passed in.
 ! it should have already been allocated by the caller.
-!---------------------------------------------------------------------
+!
+! handles " " or ' ' quoted strings, and \ escapes the next character.
+! for the name=val form, a final & sets a return flag to indicate
+! there is a following line that is part of the same context.
+!
+
+module parse_args_mod
+
+use utilities_mod, only : error_handler, E_ERR
 
 implicit none
 private
 
-! version controlled file description for error handling, do not edit
-character(len=256), parameter :: source   = &
-   "$URL$"
-character(len=32 ), parameter :: revision = "$Revision$"
-character(len=128), parameter :: revdate  = "$Date$"
-
-public :: get_args_from_string
+public :: get_args_from_string, &
+          get_name_val_pairs_from_string, &
+          get_next_arg
 
+character(len=*), parameter :: source = 'parse_args_mod.f90'
```

Comment:
i like this change, and only passing in source to the error_handler.  this seems like a job for awk or sed to do all the files in the repo - remove svn keywords and revision and revdate in the error calls.  i know tim has another issue open on this already.  i'm just cheering you guys on!